### PR TITLE
[TD-11944]<feature>: bulk pulling

### PR DESCRIPTION
**Pull request by huskar-t**

### CODE REVIEW

File: controller/rest/restful_test.go [hunk 1]
```diff
@@ -218,3 +221,330 @@ func TestRowLimit(t *testing.T) {
 	assert.Equal(t, map[string]interface{}{"table": "t1"}, result.Data[0][14])
 	assert.Equal(t, "t1", result.Data[0][15])
 }
+
+func TestWebsocket(t *testing.T) {
```
Comment (sangshuduo) [inline controller/rest/restful_test.go:225]: better to have a 'ws_test.go' and use it to test ws.go.

File: controller/rest/ws.go [hunk 2]
```diff
@@ -0,0 +1,663 @@
...
+	"github.com/taosdata/taosadapter/tools/web"
+)
+
+const TaosSessionKey = "taos"
+const (
+	WSConnect    = "conn"
+	WSQuery      = "query"
+	WSFetch      = "fetch"
+	WSFetchBlock = "fetch_block"
+	WSFetchJson  = "fetch_json"
```
Comment (sangshuduo) [inline controller/rest/ws.go:36]: is json function still available?

**Review by sangshuduo — COMMENTED** [review]
see my comments

**Comment by huskar-t** [comment]
> see my comments

resolved

**Review by sangshuduo — APPROVED** [review]
LGTM

**Review by zitsen — APPROVED** [review]
LGTM

**Pull request merged**